### DISCUSSION TITLE
Trying to convert selenium to seleniumbase

### DISCUSSION THREAD

**Discussion by ScottCov** [seleniumbase/SeleniumBase#3121] [General]
Hi-I am trying to convert the following from selenium to seleniumbase:

```
rows=driver.find_elements(By.XPATH,'//*[@id="results-table"]/tbody/tr[not(@class="jqgfirstrow")]')
for row in rows:
        row1=row.find_element(By.XPATH,'./td[1]/span')
```
Tried the following:
```
rows=driver.find_elements('[id="results-table"] tbody tr:not([class="jqgfirstrow"])', by="css selector", limit=0)
for row in rows:
        row1=row.find_element('td:first-of-type span')
```
getting error:
selenium.common.exceptions.InvalidArgumentException: Message: invalid argument: invalid locator

Can anyone steer me in the right direction?

Thanks

Scott

Debian 12
Python 3.10.8

**Comment by mdmintz** [comment]
`find_elements()` always returns a list of `WebElement` objects. `WebElement` uses the Selenium API, which requires specifying the selector `by` type before the selector.

So basically it would look like this: (You get rows using the SeleniumBase call, which doesn't need the `by`. Then you loop through the WebElement objects, which do need the `by` before the selector.

```python
rows = driver.find_elements('[id="results-table"] tbody tr:not([class="jqgfirstrow"])')
for row in rows:
    row1 = row.find_element(By.CSS_SELECTOR, "td:first-of-type span")
```

**Reply by ScottCov** [reply]
Thanks Mike!

**Reply by ScottCov** [reply]
Hi Mike-

I am trying to do the same again as above but instead check if an attribute exists....I tried
`if row1.is_attribute_present(By.CSS_SELECTOR,'td[role="gridcell"]:nth-child(2) span p a','onclick'):`

AttributeError: 'WebElement' object has no attribute 'is_attribute_present'

Do I need to do something like:
```
 def isAttribtuePresent(self,element,attribute):
      result = False
      try:
          strvalue = element.getAttribute(attribute)
          if (strvalue ==''):
              result = True
      except:
          pass
      
      return result

element=row1.find_element(By.CSS_SELECTOR,'td[role="gridcell"]:nth-child(2) span p a')
if self.isAttribtuePresent(element,'onclick'):
```
?

**Reply by mdmintz** [reply]
Probably. Once you have a WebElement, that's the Selenium API, not the SeleniumBase one.
